linked_list/linked_list.py

```diff
@@ -2,78 +2,140 @@
 # Defines a node in the singly linked list
 class Node:
 
-    def __init__(self, value, next_node = None):
+    def __init__(self, value, next_node = None,previous = None):
         self.value = value
         self.next = next_node
+        self.previous = previous
 
 # Defines the singly linked list
 class LinkedList:
     def __init__(self):
       self.head = None # keep the head private. Not accessible outside this class
-
+    #   self.tail = None
     # returns the value in the first node
     # returns None if the list is empty
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(1)
+    # Space Complexity: O(1)
     def get_first(self):
-        pass
-
+        if self.head == None:

```

Comment:
Small code style note: generally prefer to use `is` when comparing to what we call singleton values like`True`, `False`, and `None` as sometimes the `==` operator can be redefined with the `__eq__` method and occasionally lead to unexpected behavior with these values.

```suggestion
        if self.head is None:

```